src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs

```diff
@@ -972,6 +975,111 @@ internal string GetPropertyRawText()
             return _parent.GetPropertyRawValueAsString(_idx);
         }
 
+        /// <summary>
+        /// Compares the string text to the unescaped JSON token value in the source and returns true if they match.
+        /// </summary>
+        /// <param name="text">The text to compare against.</param>
+        /// <returns>True if the JSON token value in the source matches the look up text.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to find a text match on a JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>).
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <remarks>
+        /// If the look up text is invalid or incomplete UTF-16 text (i.e. unpaired surrogates), the method will return false
```

Comment:
>  If the look up text is invalid or incomplete UTF-16 text (i.e. unpaired surrogates), the method will return false

> Since this is JsonElement, there's no requirement the input is UTF-8 (that's an implementation detail).

The comment wasn't about the original JSON input though, it was about the look up text that's the parameter to the `ValueEquals` method. You don't think that is relevant info to add?

Reply:
Regarding unpaired surrogates: No, it doesn't seem worth mentioning.

If the source is UTF-8 (currently it always is) then it can never produce an unpaired surrogate, so it returns false because it doesn't match.  It's the same as saying that it doesn't match when it... doesn't match.  (If it threw, that'd be different).

If the source is UTF-16(LE) (theoretical future) then either the source is validated and threw, so an unpaired surrogate test is the same as the UTF-8 case, or it isn't, and could match under ordinal rules.

So the comment (to me) is at best non-informative, and at worst (in the future) wrong.

Reply:
(Alternatively, it's just that I think it's already covered under "ordinal comparison")